### PR TITLE
If there are no open files, do not schedule ensureProjectForOpenFiles

### PR DESCRIPTION
**Pull request by sheetalkamat**
Found while working on #39656

### CODE REVIEW

File: src/server/editorServices.ts [hunk 1]
```diff
@@ -930,6 +930,8 @@ namespace ts.server {
 
         /*@internal*/
         delayEnsureProjectForOpenFiles() {
+            // If no open files or no external project, do not schedule
+            if (!this.openFiles.size && !this.externalProjects.length && !this.externalProjectToConfiguredProjectMap.size) return;
```
Comment (amcasey) [inline src/server/editorServices.ts:934]: I understand why we skip the work if there are no open files, but not why we're looking at external projects.  Can you please explain?

Comment (sheetalkamat) [reply src/server/editorServices.ts:934]: You are right.. this must be friday evening brain fog.. external projects can create configured projects not inferred projects so this doesnt make sense. i am updating the test case..

**Review by amcasey — APPROVED** [review]
The `checkTimeoutQueueLengthAndRun` change looks unrelated.  Otherwise, LGTM.

**Comment by sheetalkamat** [comment]
> The checkTimeoutQueueLengthAndRun change looks unrelated. 

This shows that we are not scheduling ensuring project for open files when there are no open files. In all those test cases there are no open files. Just external projects

**Pull request merged**